python/test/unit/language/test_core.py

```diff
@@ -390,6 +390,7 @@ def _mod_operation_ill_conditioned(dtype_x, dtype_y) -> bool:
         ('int64', 'float16'),
         ('int64', 'float32'),
         ('int64', 'float64'),
+        ('uint16', 'float32'),
```

Comment:
it errored by a small percentage, so just bump the tolerance a bit

Reply:
ahh, that makes sense. I was wondering why the tests on the different archs became divergent. Thank you!

Reply:
From further testing locally on a 3090, it seems that the difference between triton and the ref arrays are much larger. The single mismatched element that is making the test fail has the following values:
`triton = -5.173683166503906e-05` and `numpy = 1.3480271`

Almost seems like a type conversion problem.

If I add the ill conditioned pair (uint16, float32) then it fails on the h100 seems rather odd since it fails when the ill-conditioned check isn't present.

Reply:
Yes, this test is completely cursed. It involves an incredible amount of UB that happens to work by chance in most cases, but sometimes it breaks, like here lol

Would you be keen on taking on the side-quest of improving the test, so that we don't sample numbers that are too large if we are going to cast them to a dtype where they don't fit? In particular, you'd get rid of `_mod_operation_ill_conditioned`, and you'd just set a `max` value for the sampling depending on both of the dtypes. I think this would fix this issue + would get rid of the horrible workaround of `_mod_operation_ill_conditioned`.

Reply:
Ha, that is incredible, I can definitely take on the side-quest! I appreciate the direction, shouldn't be terribly difficult.

Reply:
Apologies for the delay, planned on working on this earlier. Pushed up changes for this to work on all bit-width integral types. I may change it to keep the type range if the bit-widths are equal but things get weird with uints since they have a higher range.